addons/point_of_sale/static/src/app/models/data_service.js

```diff
@@ -0,0 +1,282 @@
+            this.network.offline = false;
+            this.network.warningTriggered = false; // Avoid the display of the offline popup multiple times
+        }
+
+        this.syncData();
+    }
+
+    resetUnsyncQueue() {
+        this.network.unsyncData = [];
+        this.setOnline();
+    }
+
+    async initData() {
+        const modelClasses = {};
+        const response = await this.orm.call("pos.session", "load_data", [
+            odoo.pos_session_id,
+            PosData.modelToLoad,
+        ]);
+
+        for (const posModel of registry.category("pos_available_models").getAll()) {
+            modelClasses[posModel.pythonModel] = posModel;
+        }
+
+        const [models, records] = createRelatedModels(
+            response.relations,
+            modelClasses,
+            INDEXED_DB_NAME
+        );
+
+        this.fields = response.fields;
+        this.relations = response.relations;
+        this.custom = response.custom;
+        this.models = models;
+        this.models.loadData(response.data, this.modelToLoad);
+
+        for (const [name, model] of Object.entries(records)) {
+            this[name] = Object.values(model);
+        }
+
+        this.network.loading = false;
+    }
+
+    async execute({
+        type,
+        model,
+        ids,
+        values,
+        method,
+        queue,
+        args = [],
+        kwargs = {},
+        fields = [],
+        options = [],
+    }) {
+        this.network.loading = true;
+
+        try {
+            let result = true;
+
+            if (fields.length === 0) {
+                fields = this.fields[model];
+            }
+
+            switch (type) {
+                case "write":
+                    result = await this.orm.write(model, ids, values);
+                    break;
+                case "delete":
+                    result = await this.orm.delete(model, ids);
+                    break;
+                case "call":
+                    result = await this.orm.call(model, method, args, kwargs);
+                    break;
+                case "read":
+                    queue = false;
+                    result = await this.orm.read(model, ids, fields, {
+                        ...options,
+                        load: false,
+                    });
+                    break;
+                case "search_read":
+                    queue = false;
+                    result = await this.orm.searchRead(model, args, fields, {
+                        ...options,
+                        load: false,
+                    });
+            }
+
+            if (type === "create") {
+                const response = await this.orm.create(model, values);
+                values[0].id = response[0];
+                result = values;
+            }
+
+            if (this.models[model] && LOADED_ORM_METHODS.includes(type)) {
+                const data = this.models.loadData({ [model]: result });
+                result = data;
+            }
```

Comment:
we should only return the special structure `{ [model]: records }` when we are calling `callRelated`. But for searchRead, we should just directly return the records. The idea is that the API should be similar to `searchRead` from `web.orm`.